### PR TITLE
CeramicStudio/web-playground#29: Quickstart Upgrade

### PR DESCRIPTION
**Pull request by Sterahi**
Changes:
- Old quickstart was moved to exist under the CLI Heading
- New Quickstart focused on JavaScript client has replaced the old Quickstart.

### CODE REVIEW

File: docs/build/quick-start.md [hunk 1]
```diff
@@ -1,303 +1,138 @@
-# Quick start
-Learn the basics by setting up and interacting with the [Ceramic CLI](./cli/installation.md). This tutorial serves as a simple introduction to Ceramic concepts.
+# Quickstart
+Learn the basics by setting up and interacting with the [Ceramic CLI](./cli/installation.md)
```
Comment (stbrody) [inline docs/build/quick-start.md:2]:
```suggestion
Learn the basics by setting up and interacting with the [Ceramic Javascript Client](./javascript/installation.md)
```

File: docs/build/quick-start.md [hunk 2]
```diff
@@ -1,303 +1,138 @@
...
+## **1. Install the Client**
 
-## **1. Install the CLI**
-
-Visit the [Ceramic CLI](./cli/installation.md) page for instructions on how to quickly install the CLI.
+Visit the [Ceramic Client](./javascript/installation.md) page for instructions on how to quickly install the Client.
 
 ## **2. Create a stream**
-Use the `create` command to create a new [stream](../learn/glossary.md#streams). In the example below we create a stream that uses the [TileDocument StreamType](../streamtypes/tile-document/overview.md). Note that *TileDocument* is the only [StreamType](../learn/glossary.md#streamtypes) that can currently be created by the Ceramic CLI.
+Creating streams is dependent on the StreamType of the record. More details can be found in our [StreamTypes Overview](../streamtypes/overview.md). A basic sample can be found below.
```
Comment (stbrody) [inline docs/build/quick-start.md:12]:
```suggestion
Creating streams is dependent on the StreamType of the record. More details can be found in our [StreamTypes Overview](../streamtypes/overview.md). A basic sample using the [TileDocument StreamType](../streamtypes/tile-document/overview.md) can be found below.
```

File: docs/build/quick-start.md [hunk 3]
```diff
@@ -1,303 +1,138 @@
...
-    $ ceramic show kjzl6cwe1jw147ww5d8pswh1hjh686mut8v1br10dar8l9a3n1wf8z38l0bg8qa
-    ```
-    
-    !!! quote ""
-        You should use your StreamID instead of the StreamID included here.
+    ```JavaScript
+    import CeramicClient from '@ceramicnetwork/http-client'
 
-=== "Output"
+    const ceramic = new CeramicClient("https://gateway-clay.ceramic.network")
```
Comment (stbrody) [inline docs/build/quick-start.md:32]: Here we show how to set up the client but above for the "create a stream" example we do not.  Whichever we do we should be consistent.

File: docs/build/quick-start.md [hunk 4]
```diff
@@ -1,303 +1,138 @@
...
 
-    ```bash
-    {
-        "Foo": "Bar"
-    }
+    const streamId = "kjzl6cwe1jw147ww5d8pswh1hjh686mut8v1br10dar8l9a3n1wf8z38l0bg8qa"
```
Comment (stbrody) [inline docs/build/quick-start.md:34]: We're inconsistent throughout this page on when we use concrete examples with real values, like we do here, and when we use placeholder variables, like in the "create a stream" example above where we use `content`, `metadata`, and `opts` without showing what they actually would look like.  Again, whatever we do we should be consistent.

File: docs/build/quick-start.md [hunk 5]
```diff
@@ -1,303 +1,138 @@
...
-    }
+    await doc.update({foo: 'baz'}, {tags: ['baz']})
     ```
-
-??? info "More options"
-    Currently you can change *content*, *controllers*, and *schema* using the CLI. Run `ceramic update -h` for more information.
-
+<!-- API Reference here -->
 
 ## **5. Create a schema**
```
Comment (stbrody) [inline docs/build/quick-start.md:64]: I wonder if we should even include this schema stuff in the quick start, or just leave it in the TileDocument specific docs.  Doesn't seem as relevant as the basic CRUD ops for a quick start guide

File: mkdocs.yml [hunk 6]
```diff
@@ -30,88 +29,89 @@ theme:
...
-    - Networks: learn/networks.md
-    - Blog: learn/blog.md
-    - Explore:
-      - Explorers: explore/explorers.md
-      - Projects: explore/projects.md
-      - Sample apps: explore/sample-apps.md
-    - Advanced:
-      - Tech overview: learn/advanced/overview.md
-      - Data availability: learn/advanced/data-availability.md
+      - Welcome: learn/welcome.md
```
Comment (stbrody) [inline mkdocs.yml:32]: all these formatting changes make it hard to see what *actually* changed.  If you're going to do a pass to clean up the formatting, ideally it'd be nice to do that as a separate commit that just does the formatting so that it can be reviewed independent of any functionality changes.

**Review by stbrody — CHANGES_REQUESTED** [review]
My main comment is to clean up how we structure our examples to be consistent around how and when we use concrete examples vs just the names of the arguments.

File: docs/build/quick-start.md (hunk 4 above)
Comment (Sterahi) [reply docs/build/quick-start.md:34]: Honestly, I just missed this section. Will update & make it consistent :)

File: mkdocs.yml (hunk 6 above)
Comment (Sterahi) [reply mkdocs.yml:32]: I did not intend to change any of this, I think VS Code just decided to mess things up. 

The long and short of it is: 
- CLI Quickstart moved to CLI section
- New quickstart moved to where the Old Quickstart was.

**Comment by michaelsena** [comment]
Yeah I’m suggesting we move the JS Quick Start into the JavaScript section
instead of just the top level in the build category. I’ve been meaning to
make this change.

We want to reserve the top level section for some developer guidance we
will write in the future.

File: docs/build/quick-start.md (hunk 5 above)
Comment (Sterahi) [reply docs/build/quick-start.md:64]: My argument for including it is just to streamline the flow to the TileDocument docs. It's never a bad thing to have more links to the docs in my eyes.

Comment (Sterahi) [reply docs/build/quick-start.md:64]: Ignore this, I thought there was somewhere else in the docs that had an API Reference. I was wrong so this isn't going to stay.

File: docs/build/cli/quick-start.md [hunk 7]
```diff
@@ -1,5 +1,5 @@
-# Quick start
-Learn the basics by setting up and interacting with the [Ceramic CLI](./cli/installation.md). This tutorial serves as a simple introduction to Ceramic concepts.
+# Quickstart
+Learn the basics by setting up and interacting with the [Ceramic CLI](./javascript/installation.md). This tutorial serves as a simple introduction to Ceramic concepts.
```
Comment (stbrody) [inline docs/build/cli/quick-start.md:2]:
```suggestion
Learn the basics by setting up and interacting with the [Ceramic CLI](./cli/installation.md). This tutorial serves as a simple introduction to Ceramic concepts.
```

File: docs/build/javascript/quick-start.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
+# Quickstart
+Learn the basics by setting up and interacting with the [Ceramic CLI](./cli/installation.md)
```
Comment (stbrody) [inline docs/build/javascript/quick-start.md:2]:
```suggestion
Learn the basics by setting up and interacting with the [Ceramic Javascript Client](./javascript/installation.md)
```

Comment (stbrody) [reply docs/build/javascript/quick-start.md:2]: I don't know if I messed up my suggested edits last time around or if something got re-arranged here

File: docs/build/javascript/quick-start.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+!!! warning ""
+    **Want an even faster way to try Ceramic? Visit the [Playground Demo App](https://playground.ceramic.dev){:target="_blank"} to test the full stack of Ceramic Components in the browser.
+
+## **1. Install the Client**
+
+Visit the [Ceramic Client](./javascript/installation.md) page for instructions on how to quickly install the Client.
+
+## **2. Create a stream**
+Creating streams is dependent on the StreamType of the record. More details can be found in our [StreamTypes Overview](../streamtypes/overview.md). A basic sample can be found below.
```
Comment (stbrody) [inline docs/build/javascript/quick-start.md:12]:
```suggestion
Creating streams is dependent on the StreamType. More details can be found in our [StreamTypes Overview](../streamtypes/overview.md). A basic sample using the [TileDocument StreamType](../streamtypes/tile-document/overview.md) can be found below.
```

File: docs/build/javascript/quick-start.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+
+=== "Command"
+
+    ```JavaScript
+    import { TileDocument } from '@ceramicnetwork/stream-tile'
+
+    const schemaId = 'k3y52l7qbv1fry1fp4s0nwdarh0vahusarpposgevy0pemiykymd2ord6swtharcw' // this is our Basic Profile Schema.
+    
+    const content = {
```
Comment (stbrody) [inline docs/build/javascript/quick-start.md:21]: I don't think this will pass validation of the basic profile schema.  If we're going to use a real concrete schema in the example we should make sure it actually works with the example data

Comment (stbrody) [reply docs/build/javascript/quick-start.md:21]: I'm not sure how big is the minimal profile that will be accepted by the Basic Profile schema.  If it's small you could update the test data to match the schema, but if it's large I also think it'd be fine to just not set a schema in this example.

File: docs/build/javascript/quick-start.md [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+    ```
+
+## **3. Query a stream**
+Use the `loadStream()` function to query the current [state](../learn/glossary.md#state) of a stream. You will need to provide it's *StreamID*
+
+=== "Command"
+
+    ```JavaScript
+    const streamId = "kjzl6cwe1jw147ww5d8pswh1hjh686mut8v1br10dar8l9a3n1wf8z38l0bg8qa"
+    const stream = await ceramic.loadStream(streamId)
```
Comment (stbrody) [inline docs/build/javascript/quick-start.md:42]:
```suggestion
    const stream = await TileDocument.load(streamId)
```

Comment (stbrody) [reply docs/build/javascript/quick-start.md:42]: Using the StreamType-specific load method will let you use StreamType specific functionality.  For example, the way streams get updated is streamtype-specific, so if you want to be able to access the TileDocument's `.update()` method to update the stream after loading it, you need to specify the type.  The default `Stream` type that's returned by `loadStream()` doesn't include any update functionality (or any other streamtype-specific accessors)

File: docs/build/javascript/quick-start.md [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+      ]
+    }
+    const metadata = {
+      controllers: [ceramic.did.id] // this will set yourself as the controller of the schema
+    }
+    const rewardSchema = await TileDocument.create(ceramic, schema, metadata)
+    ```
+
+## **6. Create a TileDocument stream that uses a schema**
+First, use the `rewardSchema.commitId.toString()` to list the [commitIDs](../learn/glossary.md#commitid) contained in the schema stream. When creating a TileDocument that usees this schema need to use a commitID instead of the StreamID. This is to enforce that we are using a specific version of the schema since the schema stream is mutable and can be updated.
```
Comment (stbrody) [inline docs/build/javascript/quick-start.md:96]: `rewardSchema.commitId.toString()` just gets the CommitID of the current version, it doesn't list all CommitIDs for the history of the stream.  To do that you would do `rewardSchema.allCommitIds()`.

I think for this example just taking the current CommitID is probably fine

Comment (stbrody) [reply docs/build/javascript/quick-start.md:96]:
```suggestion
First, use the `rewardSchema.commitId.toString()` to get the current [commitID](../learn/glossary.md#commitid) of the schema stream. When creating a TileDocument that usees this schema need to use a commitID instead of the StreamID. This is to enforce that we are using a specific version of the schema since the schema stream is mutable and can be updated.
```

File: docs/build/javascript/quick-start.md [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+        title: 'Hello',
+        message: 'world!'
+      }, {
+        controllers: [ceramic.did.id],
+        family: 'Rewards',
+        schema: rewardSchema.commitId.toString(),
+      })
+    ```
+
+## **7. Query the stream you created**
```
Comment (stbrody) [inline docs/build/javascript/quick-start.md:111]:
```suggestion
## **7. Inspect the state of the stream you created**
```

**Review by stbrody — COMMENTED** [review]
looks like maybe some of the comments from the last round got lost in the reshuffling of files around?

File: docs/build/javascript/quick-start.md (hunk 10 above)
Comment (Sterahi) [reply docs/build/javascript/quick-start.md:21]: Oops, wrote the content before adding the schema. 

Regarding not needing the schema, I thought the same thing initially but decided to use the basic schema so that someone following along could actually create their own stream easily. I'd rather provide all of the useful information where it's needed (unless it's explained in a previous step).

File: docs/build/javascript/quick-start.md [hunk 14]
```diff
@@ -0,0 +1,146 @@
+# Quickstart
+Learn the basics by setting up and interacting with the [Ceramic Javascript Client](./javascript/installation.md).
+
+!!! warning ""
+    **Want an even faster way to try Ceramic? Visit the [Playground Demo App](https://playground.ceramic.dev){:target="_blank"} to test the full stack of Ceramic Components in the browser.
```
Comment (stbrody) [inline docs/build/javascript/quick-start.md:5]:
```suggestion
    **Want an even faster way to try Ceramic? Visit the [Playground Demo App](https://playground.ceramic.dev){:target="_blank"} to test the full stack of Ceramic components in the browser.
```

File: docs/build/javascript/quick-start.md [hunk 15]
```diff
@@ -0,0 +1,146 @@
...
+
+## **2. Create a stream**
+Creating streams is dependent on the StreamType. More details can be found in our [StreamTypes Overview](../streamtypes/overview.md). A basic sample using the [TileDocument StreamType](../streamtypes/tile-document/overview.md) can be found below.
+
+=== "Command"
+
+    ```JavaScript
+    import { TileDocument } from '@ceramicnetwork/stream-tile'
+
+    const schemaId = 'k3y52l7qbv1fry1fp4s0nwdarh0vahusarpposgevy0pemiykymd2ord6swtharcw' // this is our Basic Profile Schema.
```
Comment (stbrody) [inline docs/build/javascript/quick-start.md:19]:
```suggestion
    const schemaId = 'k3y52l7qbv1fry1fp4s0nwdarh0vahusarpposgevy0pemiykymd2ord6swtharcw' // this is the IDX Basic Profile Schema.
```

File: docs/build/javascript/quick-start.md (hunk 10 above)
Comment (stbrody) [reply docs/build/javascript/quick-start.md:21]: for the record, schemas are optional in TileDocuments so you don't *need* to set a schema to create a Stream.  But I see you updated the content to match the basic profile schema, and that seems fine for this example.

File: docs/build/javascript/quick-start.md [hunk 16]
```diff
@@ -0,0 +1,146 @@
...
+    const streamId = doc.id.toString()
+    ```
+
+## **3. Query a stream**
+Use the `load()` function to query the current [state](../learn/glossary.md#state) of a stream. You will need to provide it's *StreamID*.
+
+=== "Command"
+
+    ```JavaScript
+    const streamId = "kjzl6cwe1jw147ww5d8pswh1hjh686mut8v1br10dar8l9a3n1wf8z38l0bg8qa"
```
Comment (stbrody) [inline docs/build/javascript/quick-start.md:43]:
```suggestion
    const streamId = "kjzl6cwe1jw147ww5d8pswh1hjh686mut8v1br10dar8l9a3n1wf8z38l0bg8qa"  // This is just an example StreamID, you will need to use a StreamID relevant to your app.
```

Comment (stbrody) [reply docs/build/javascript/quick-start.md:43]: This shouldn't be necessary but unfortunately we've already gotten people trying to follow our examples **exactly** and hitting issues. :facepalm:

File: docs/build/javascript/quick-start.md [hunk 17]
```diff
@@ -0,0 +1,146 @@
...
+      ]
+    }
+    const metadata = {
+      controllers: [ceramic.did.id] // this will set yourself as the controller of the schema
+    }
+    const rewardSchema = await TileDocument.create(ceramic, schema, metadata)
+    ```
+
+## **6. Create a TileDocument stream that uses a schema**
+First, use the `rewardSchema.commitId.toString()` to get the current [commitIDs](../learn/glossary.md#commitid) of the schema stream. When creating a TileDocument that uses this schema need to use a commitID instead of the StreamID. This is to enforce that we are using a specific version of the schema since the schema stream is mutable and can be updated.
```
Comment (stbrody) [inline docs/build/javascript/quick-start.md:98]:
```suggestion
First, use the `rewardSchema.commitId.toString()` to get the current [CommitID](../learn/glossary.md#commitid) of the schema stream. When creating a TileDocument that uses this schema you need to use a CommitID instead of the StreamID. This is to enforce that we are using a specific version of the schema since the schema stream is mutable and can be updated.
```

File: docs/build/javascript/quick-start.md [hunk 18]
```diff
@@ -0,0 +1,146 @@
...
+        message: 'world!'
+      }, {
+        controllers: [ceramic.did.id],
+        family: 'Rewards',
+        schema: rewardSchema.commitId.toString(),
+      })
+    ```
+
+## **7. Inspect the state of the stream you created**
+Use `reward.state` to query the state of the TileDocument we just created. We can see that the schema is set to the correctID.
```
Comment (stbrody) [inline docs/build/javascript/quick-start.md:114]:
```suggestion
Use `reward.state` to query the state of the TileDocument we just created. We can see that the schema is set to the correct CommitID.
```

File: mkdocs.yml [hunk 19]
```diff
@@ -150,37 +161,66 @@ extra_css:
...
-            'build/queries.md': 'build/javascript/queries.md'
-            'build/pinning.md': 'build/javascript/pinning.md'
-            'reference/http-api.md': 'build/http/api.md'
-            'tools/hosted-nodes/community-nodes.md': 'run/nodes/community-nodes.md'
-            'tools/hosted-nodes/node-providers.md': 'run/nodes/node-providers.md'
-            'tools/identity/idx.md': 'tools/idx/overview.md'
-            'tools/identity/identitylink.md': 'tools/identitylink/overview.md'
-            'tools/identity/self-id.md': 'tools/self-id/overview.md'
-    - search
+  # - redirects:
```
Comment (stbrody) [inline mkdocs.yml:164]: Can we remove the old versions instead of leaving them here commented out?

**Review by stbrody — APPROVED** [review]
LGTM with some small final suggestions

File: docs/build/javascript/quick-start.md (hunk 16 above)
Comment (Sterahi) [reply docs/build/javascript/quick-start.md:43]: This actually is a live stream and not a demo stream. All demo streams are concatenated so that they are not useable specifically for this reason!

File: mkdocs.yml (hunk 19 above)
Comment (Sterahi) [reply mkdocs.yml:164]: Absolutely. Working with 2 branches in the same repo messed me up a bit. All the changes were updated on the IDX branch, but I wanted to get this updated to not be messed up formatting wise.

File: docs/build/javascript/quick-start.md [hunk 20]
```diff
@@ -0,0 +1,146 @@
...
+    
+    const content = {
+      name: 'Your Name',
+      emoji: '✌🏻',
+      description: 'Something about yourself'
+    }
+    const metadata = {
+      controllers: [ceramic.did.id],
+      family: 'Basic Profile',
+      schema: schemaId
```
Comment (oed) [inline docs/build/javascript/quick-start.md:29]: Do we really want to demonstrate the use of an IDX schema / definition as the first example here? Wouldn't it be more easy to just create an hello world stream?

**Review by oed — COMMENTED** [review]
Looks good, but I would simplify the first example!

File: docs/build/javascript/quick-start.md (hunk 20 above)
Comment (Sterahi) [reply docs/build/javascript/quick-start.md:29]: It would be easier, but I feel like it would create a less holistic experience. I think that this example is easily parsable & would also work well as a reference for anyone just starting out.

Comment (oed) [reply docs/build/javascript/quick-start.md:29]: But isn't IDX covered in the IDX specific quickstart?

Comment (Sterahi) [reply docs/build/javascript/quick-start.md:29]: That's true, I think I got this confused in my head. For some reason, I thought that creating a stream was after querying a stream. Will update this example though.

Comment (Sterahi) [reply docs/build/javascript/quick-start.md]: I now remember why I had it set up how it was. We cover creating a stream later in the documentation and the schema must be a commitId. So it just made more sense to not duplicate the content.

File: docs/build/javascript/quick-start.md [hunk 21]
```diff
@@ -0,0 +1,133 @@
...
+Creating streams is dependent on the StreamType. More details can be found in our [StreamTypes Overview](../streamtypes/overview.md). A basic sample using the [TileDocument StreamType](../streamtypes/tile-document/overview.md) can be found below.
+
+=== "Command"
+
+    ```JavaScript
+    import { TileDocument } from '@ceramicnetwork/stream-tile'
+
+    const doc = await TileDocument.create(ceramic, {hello: 'world'})
+
+    const streamId = doc.id.toString()
```
Comment (oed) [inline docs/build/javascript/quick-start.md:21]: Maybe also add a `console.log(doc.content)` here to quickly show where the data end up?

File: docs/build/javascript/quick-start.md [hunk 22]
```diff
@@ -0,0 +1,133 @@
...
+    ```
+
+## **3. Query a stream**
+Use the `load()` function to query the current [state](../learn/glossary.md#state) of a stream. You will need to provide it's *StreamID*.
+
+=== "Command"
+
+    ```JavaScript
+    const streamId = "kjzl6cwe1jw147ww5d8pswh1hjh686mut8v1br10dar8l9a3n1wf8z38l0bg8qa"
+    const stream = await TileDocument.load(streamId)
```
Comment (oed) [inline docs/build/javascript/quick-start.md:31]: We call the variable `stream` here, but above and below we use `doc`.

Comment (oed) [reply docs/build/javascript/quick-start.md:31]: Might make sense to do `console.log(doc.content)` here as well!

**Review by oed — APPROVED** [review]
This should be fine to merge after my comments are addressed!
